custom_md_loader/classes/MetadataApexApiLoader.cls

```diff
@@ -0,0 +1,230 @@
+	            
+	            	if(mappingInfo.getCSToMDT_fieldMapping().get(fieldName).endsWith('__c')){
+	        			Metadata.CustomMetadataValue cmv = new Metadata.CustomMetadataValue();
+	                	cmv.field = mappingInfo.getCSToMDT_fieldMapping().get(fieldName);
+	                	if(descCSFieldResult.getType().name() == 'DATETIME') {
+	                	
+	                		if(csRecord.get(fieldName) != null) {
+								Datetime dt = DateTime.valueOf(csRecord.get(fieldName));
+								String formattedDateTime = dt.format('yyyy-MM-dd\'T\'HH:mm:ss.SSSZ'); 
+								cmv.value = csRecord.get(formattedDateTime);
+							}
+							else {
+								cmv.value = null;
+							}
+							
+						}
+						else{
+		            		cmv.value = csRecord.get(fieldName);
+	            		}
+	                	
+	                	customMetadataRecord.values.add(cmv);
+	            	}
+	        	}
+	            records.add(customMetadataRecord);
+	        }
+	            
+			
+		    callback.setMdDeployStatus(mdDeployStatus);
+		    
+			Metadata.DeployContainer deployContainer = new Metadata.DeployContainer();
+			for(Metadata.CustomMetadata record : records) {
+	        	deployContainer.addMetadata(record);
+	        }
+			
+			// Enqueue custom metadata deployment
+	    	Id jobId = Metadata.Operations.enqueueDeployment(deployContainer, callback);
+	    	jobId1 = jobId;
+	    	
+	    	mdDeployStatus.setJobId(jobId);
+	    	
+	    	System.debug('jobId-->' + jobId);
+	    	System.debug('mdDeployStatus-->' + mdDeployStatus);
+	    	System.debug('mdDeployStatus.getJobId()-->' + mdDeployStatus.getJobId());
+	    }
+	    catch(Exception e) {
+	    	System.debug('MetadataWrapperApiLoader.Error Message=' + e.getMessage());
+    		List<MetadataResponse.Message> messages = new List<MetadataResponse.Message>();
+    		messages.add(new MetadataResponse.Message(100, e.getMessage()));
+        
+    		response.setIsSuccess(false);
+    	    response.setMessages(messages);  
+        }
+    }
+    
+    
+    	
+	public class MetadataDeployStatus {
+		public Id jobId {get;set;}
+		public Metadata.DeployStatus deployStatus {get;set;}
+		public boolean success {get;set;}
+	
+		public MetadataDeployStatus() {}
+		
+		public Id getJobId() {
+			return this.jobId;
+		}
+		public void setJobId(Id jobId) {
+			this.jobId = jobId;
+		}
+		
+		public Metadata.DeployStatus getDeployStatus() {
+			return this.deployStatus;
+		}
+		public void setDeployStatus(Metadata.DeployStatus deployStatus) {
+			System.debug('setDeployStatus deployStatus ===>'+ deployStatus);
+			this.deployStatus = deployStatus;
+		}
+		
+		public boolean getSuccess() {
+			return this.success;
+		}
+		public void setSuccess(boolean success) {
+			System.debug('setDeployStatus success ===>'+ success);
+			this.success = success;
+		}
+	
+	}
+	    
+	public class MetadataDeployCallback implements Metadata.DeployCallback {

```

Comment:
We talked a little bit about ways to use this to inform the user of the deploy result. Could you leave a TODO at the right point to do something like polling or emailing the user, so we can follow up on it later?

Reply:
agreed.